### DISCUSSION TITLE
Uncaught Error: WebSocket close with status code 1006

### DISCUSSION THREAD

**Discussion by Miky88** [coder/code-server#2698] [Q&A] [answered]
[Help Wanted - not a bug, i couldn't be able to join the Slack workspace]

I managed to run the code-server on my VPS running nginx web server, proxied through Cloudflare free plan with the following DNS configuration:

## DNS configuration:
![immagine](https://user-images.githubusercontent.com/44903677/107141768-55449400-692b-11eb-8c0f-1e98b9553e95.png)

## Firefox logs: 
![immagine](https://user-images.githubusercontent.com/44903677/107141549-aa7fa600-6929-11eb-9ed0-c451239775ba.png)

The page doesn't load, it works just fine with the "--link" option but I can't figure out why is it not working with my custom domain setup. Not a client problem since my friends can't connect too, hope it's not CF...

- Web Browser: Firefox, 85.0.1 (64 bit)
- Local OS: Windows 10 (64 bit)
- Remote OS: Ubuntu 20.04.2 LTS
- Remote Architecture: x86_64
- `code-server --version`: 3.8.0 c4610f7829701aadb045d450013b84491c30580d

**Comment by jsjoeio** [comment]
Sorry to hear you're having issues. Let's see what we can do to help you!

> [Help Wanted - not a bug, i couldn't be able to join the Slack workspace]

Why weren't you able to join the Slack workspace? cc @bpmct 

~~Since this is not a bug or an issue, can you please ask this a Question under Discussion? [Link](https://github.com/cdr/code-server/discussions/new?category_id=22503114)~~

**Comment by bpmct** [comment]
Slack link works fine for me. To confirm, were you using https://cdr.co/join-community?

**Reply by Miky88** [reply]
![immagine](https://user-images.githubusercontent.com/44903677/107259730-a6dd4380-6a3d-11eb-87b3-af3992d4cf7a.png)
Invitation expired
You appear to be trying to accept an invitation but the link has been disabled.
Contact the person who initially invited you to get a new link or manual invitation.

**Reply by Miky88** [reply]
Yes, i was using that link.

**Reply by bpmct** [reply]
hmm... i updated it again, but you can also try https://join.slack.com/t/coder-com/shared_invite/zt-irpkf25c-mk1g2aSqHVb_IFoEAynr9Q

**Reply by Miky88** [reply]
uh? still not working... maybe something with my account..?

**Reply by bpmct** [reply]
so strange... send me an email ben [at ] coder.com I'll send you an invite.

**Comment by bpmct** [comment, marked as answer]
@Miky88 I believe there are some custom proxy rules needed for NGINX forwarding, including the websocket.

https://github.com/cdr/code-server/blob/v3.8.1/docs/guide.md#nginx

Also, did you set up a certificate on your server or are you just using CloudFlare's "Flexible" SSL?

**Reply by Miky88** [reply]
We're just using CloudFlare's "Flexible" SSL.

**Reply by Miky88** [reply]
yes it's working now, i followed guide steps. Thank you!
